app/controllers/questions.rb

```diff
@@ -1,16 +1,37 @@
-get '/questions/new' do
+get '/categories/:cid/questions/new' do
+	if logged_in?
+		@category = Category.find(params[:cid])
+		@all_categories = Category.order(:title)
+		erb :'/questions/new'
+	else 
+		redirect '/users/login'
```

Comment:
it would be nice to add in an error here about why they are being redirected - even if it's just hard coded